### PR TITLE
Protect against invalid input

### LINKED ISSUE #164: Unable to Remove Selected Body
The remove button does not have an effect.

### LINKED ISSUE #166: Altering body position or velocity breaks simulation

### PR DESCRIPTION
**Pull request by aaroncameron21**
How to test:
* Select a body
* Try and type invalid input into all of the text boxes. *'e' is the only letter allowed*
* Play and pause intermittently while doing this

_______

* Closes #166
* Closes #164

### CODE REVIEW

File: partials/body-data.html [hunk 1]
```diff
@@ -3,64 +3,64 @@
...
+    <h3 class="panel-title">{{editingBody.name}}</h3>
   </div>
   <div class="panel-body">
     <form>
       <div class='form-group'>
         <label for='radius'>Radius</label>
         <div class='input-group'>
-          <input id='radius' class='form-control input-sm' 
-            ng-model="selectedBody.radius" ng-change='b.updateBody(selectedBody)'>
+          <input id='radius' type="number" class='form-control input-sm'
```
Comment (benjic) [inline partials/body-data.html]: I was looking at adding the same validation directive. The problem I found is that an empty text field is valid and will result in a `undefined` value being applied to the body.

Comment (aaroncameron21) [inline partials/body-data.html]: I believe the update function in body.js will fall back to the previous value if it receives an undefined for any attribute.

Comment (benjic) [inline partials/body-data.html]: :+1:

**Comment by benjic** [comment]
I can follow up with this last PR as soon as the merge conflicts are resolved.

**Comment by aaroncameron21** [comment]
:+1: Merge is a go

**Pull request merged**